### PR TITLE
Add service extensions after first frame event received.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Prevents a crash in the framework. 

Crash was occurring if the Performance Overlay or Hide Debug Mode Banner settings were enabled before performing a hot restart.

### CODE REVIEW

File: src/io/flutter/server/vmService/VMServiceManager.java [hunk 1]
```diff
@@ -238,6 +245,25 @@ private void onVmServiceReceived(String streamId, Event event) {
     }
   }
 
+  private void maybeAddServiceExtension(String name) {
+    if (firstFrameEventReceived) {
+      addServiceExtension(name);
+      if (pendingServiceExtensions.size() > 0) {
```
Comment (devoncarew) [inline src/io/flutter/server/vmService/VMServiceManager.java]: perhaps `!list.isEmpty()`

Comment (kenzieschmoll) [reply src/io/flutter/server/vmService/VMServiceManager.java]: Done.

**Comment by kenzieschmoll** [comment]
@stevemessick Will this be able to make it into M31?

**Comment by stevemessick** [comment]
Yep! I may push another build today, but certainly tomorrow.

**Pull request merged**